src/mixins/java/org/spongepowered/common/mixin/core/server/level/ServerPlayerMixin.java

```diff
@@ -830,6 +831,30 @@ public void sendMessage(final net.minecraft.network.chat.Component p_241151_1_,
             }
         }
     }
+
+    @Inject(method = "setPlayerInput", at = @At("HEAD"), cancellable = true)
```

Comment:
This should be a redirect on `isPassenger`, so you can avoid calling it twice, and just return `false` here to short circuit the if statement.